include/ensmallen_bits/gradient_descent/update_policies/delta_bar_delta_update.hpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file delta_bar_delta_update.hpp
+ * @author Ranjodh Singh
+ *
+ * Delta-Bar-Delta update for Gradient Descent.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_GRAIDENT_DESCENT_DELTA_BAR_DELTA_UPDATE_HPP
+#define ENSMALLEN_GRAIDENT_DESCENT_DELTA_BAR_DELTA_UPDATE_HPP
+
+namespace ens {
+
+/**
+ * DeltaBarDelta Update Policy for Gradient Descent
+ *
+ * A heuristic designed to accelerate convergence by
+ * adapting the learning rate of each paramenter individually.
+ *
+ * According to the Delta-Bar-Delta update:
+ *
+ * - If the current gradient and the exponential moving average of
+ *   previous gradients corresponding to a parameter have the same
+ *   sign, then the learning rate for that parameter is incremented by
+ *   \f$\kappa\f$. Otherwise, it is multiplied by \f$\phi\f$ (additive
+ *   increase, multiplicative decrease).
+ *
+ * @note This implementation uses a MinimumGain parameter to set a lower
+ *     bound for the learning rate. This prevents the learing rate from
+ *     droping to zero, which can occur due to floating-point underflow.
+ *     For tasks which require extreme fine-tuning, you may need to lower
+ *     this parameter than its default value(0.01) in order to allow for 
+ *     smaller learing rates.
```

Comment:
```suggestion
 *     this parameter below its default value(0.01) in order to allow for 
 *     smaller learning rates.
```

Also, is 0.01 really a good value for this?  That is very, very far away from floating-point underflow.  If the objective is just to prevent overflow I might consider more like 1e-8 or even 1000 * numeric_limits<ElemType>::epsilon() or similar.